h2/src/test/org/h2/test/db/TestOutOfMemory.java

```diff
@@ -147,67 +148,73 @@ private static void recoverAfterOOM() throws InterruptedException {
         }
     }
 
-    private void testUpdateWhenNearlyOutOfMemory() throws SQLException, InterruptedException {
+    private void testUpdateWhenNearlyOutOfMemory() throws Exception {
         if (config.memory) {
             return;
         }
-        recoverAfterOOM();
         deleteDb("outOfMemory");
-        Connection conn = getConnection("outOfMemory;MAX_OPERATION_MEMORY=1000000");
-        Statement stat = conn.createStatement();
-        stat.execute("drop all objects");
-        stat.execute("create table stuff (id int, text varchar as space(100) || id)");
-        stat.execute("insert into stuff(id) select x from system_range(1, 3000)");
-        PreparedStatement prep = conn.prepareStatement(
-                "update stuff set text = text || space(1000) || id");
-        prep.execute();
-        stat.execute("checkpoint");
-        eatMemory(80);
-        try {
+        String url = getURL("outOfMemory", true);
+//*
+        String selfDestructor = SelfDestructor.getPropertyString(1);
+        String args[] = {System.getProperty("java.home") + File.separatorChar + "bin" + File.separator + "java",
+                        "-Xmx128m", "-XX:+UseParallelGC", // "-XX:+UseG1GC",
+                        "-cp", getClassPath(),
+                        selfDestructor,
+                        Child.class.getName(), url};
+        ProcessBuilder pb = new ProcessBuilder()
+                            .redirectOutput(ProcessBuilder.Redirect.INHERIT)
+                            .redirectError(ProcessBuilder.Redirect.INHERIT)
+                            .command(args);
+        Process p = pb.start();
+        int exitValue = p.waitFor();
```

Comment:
there is a utility method waitForTimeout() in TestKillRestart that we should probably move into a better place and use here

Reply:
Thanks, I did not know that. But it's unix only, isn't it?
Once on Java 8, it will be wait with timeout.
Anyway, since we have a cooperating child process, I was planning to use SelfDestructor, but just now realized, that actuall call  to SelfDestructor.startCountdown() was left out, will add it now.